builtin/logical/totp/path_code.go

```diff
@@ -0,0 +1,117 @@
+package totp
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	totplib "github.com/pquerna/otp/totp"
+)
+
+func pathCode(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "code/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"name": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Name of the key.",
+			},
+			"code": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "TOTP code to be validated.",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathReadCode,
+			logical.UpdateOperation: b.pathValidateCode,
+		},
+
+		HelpSynopsis:    pathCodeHelpSyn,
+		HelpDescription: pathCodeHelpDesc,
+	}
+}
+
+func (b *backend) pathReadCode(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	b.logger.Trace("totp/pathReadCode: enter")
+	defer b.logger.Trace("totp/pathReadCode: exit")
```

Comment:
This is the only function that uses tracing, perhaps we should remove it from this function?